MobileCenterPush/MobileCenterPush/MSPush.m

```diff
@@ -192,29 +215,80 @@ - (void)didFailToRegisterForRemoteNotificationsWithError:(NSError *)error {
                error.description);
 }
 
-- (BOOL)didReceiveRemoteNotification:(NSDictionary *)userInfo {
+// TODO: Review this method name.
+#if TARGET_OS_OSX
+- (BOOL)didReceiveNotification:(NSNotification *)notification {
+  return [self didReceiveUserNotification:[notification.userInfo objectForKey:NSApplicationLaunchUserNotificationKey]];
+}
+
+- (BOOL)didReceiveUserNotification:(NSUserNotification *)notification {
+  if (notification && [self didReceiveRemoteNotification:notification.userInfo fromUserNotification:YES]) {
+    NSUserNotificationCenter *center = [NSUserNotificationCenter defaultUserNotificationCenter];
+
+    // The delivered notification should be removed.
+    [center removeDeliveredNotification:notification];
+    return YES;
+  }
+  return NO;
+}
+#endif
+
+- (BOOL)didReceiveRemoteNotification:(NSDictionary *)userInfo fromUserNotification:(BOOL)userNotification {
+
+#if !TARGET_OS_OSX
+  (void)userNotification;
+#endif
   MSLogVerbose([MSPush logTag], @"User info for notification has forwarded to Push: %@", [userInfo description]);
```

Comment:
This was not changed in this PR, but I think it should say `User info for notification was forwarded...`.